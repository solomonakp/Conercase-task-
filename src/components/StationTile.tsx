import React from 'react';
import StationControl from './StationControl';
import styles from './styles/StationTile.module.scss';
import { StationProps } from '../stationsObject';
import { RootState } from '../redux/reducers/index';
import { useSelector, useDispatch } from 'react-redux';
import { toggleStation } from '../redux/reducers/radio/radioActions';

export interface StationTileProps extends StationProps {
  index: number;
}

const StationTile: React.FC<StationTileProps> = ({
  station,
  frequency,
  index,
}) => {
  const { currentStationIndex, stationLoading } = useSelector(
    (state: RootState) => state.radio
  );
  const isOpen = currentStationIndex === index ? true : false;

  const dispatch = useDispatch();

  return (
    <div className={styles.stationTile} data-testid='station-tile'>
      <StationControl isOpen={isOpen} index={index} />
      <button
        className={styles.button}
        data-testid={`tile-btn-${index}`}
        disabled={stationLoading}
        onClick={() => {
          dispatch(toggleStation(index));
        }}
        data-toggle='collapse'
        data-target={`collapse${index}`}
        aria-expanded={isOpen}
        aria-controls={`collapse${index}`}
        aria-pressed={isOpen}
      >
        <span>{station}</span>
        <span>{frequency}</span>
      </button>
    </div>
  );
};

export default StationTile;
