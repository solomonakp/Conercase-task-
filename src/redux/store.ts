import { createStore, compose, applyMiddleware, Store } from 'redux';
import thunk from 'redux-thunk';

// imports index.tx of reducers
import rootReducer from './reducers';

// createStore takes initialState as a parameter
const initialState = {};

//  middleware
const middleware = [thunk];

// dev tools extension
declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);

export default store;
