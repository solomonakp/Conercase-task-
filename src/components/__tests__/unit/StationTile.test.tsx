import React from 'react';
import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import store from '../../../redux/store';
import { Provider } from 'react-redux';
import StationTile from '../../StationTile';

describe('The <StationTile/> component', () => {
  const defaultProps = { station: 'Power FM', frequency: '93,7', index: 4 };
  const setUpTile = (props = defaultProps) =>
    render(
      <Provider store={store}>
        <StationTile {...(props as any)}></StationTile>
      </Provider>
    );

  it('renders correctly', () => {
    const { asFragment } = setUpTile();
    expect(asFragment).toMatchSnapshot();
  });

  it('should display station name and frequency', () => {
    const { getByText } = setUpTile();

    expect(getByText('Power FM')).toBeInTheDocument();

    expect(getByText('93,7')).toBeInTheDocument();
  });

  it('❌ has no accessibility validation', async () => {
    const { container } = setUpTile();

    const result = await axe(container);

    expect(result).toHaveNoViolations();
  });
});
