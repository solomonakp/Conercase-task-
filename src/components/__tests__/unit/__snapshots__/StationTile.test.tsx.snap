// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <StationTile/> component renders correctly 1`] = `[Function]`;
